jalib/jassert.cpp

```diff
@@ -63,10 +63,24 @@ static int errConsoleFd = -1;
 // (as a lvalue) to modifying the underlying variable.
 
 static jalib::string&
-tmpDir() { static jalib::string s; return s; }
+tmpDir()
+{
+  static jalib::string *s = NULL;
+  if (s == NULL) {
+    s = new jalib::string();
+  }
+  return *s;
+}
 
 static jalib::string&
-uniquePidStr() { static jalib::string s; return s; }
+uniquePidStr()
+{
+  static jalib::string *s = NULL;
+  if (s == NULL) {
```

Comment:
Suggest adding comment:
// Technically, this is a memory leak, but `s` is static and so it happens only once.